### PR TITLE
fix: UI issues in dark mode

### LINKED ISSUE #2299: [Bug] No hover state for font sizes
### Bug Description

image

### How to Reproduce

Find a document page
Go to the top right corner
Click on `:`
Hover on a font that is not in use

### Expected Behavior

Each font option should have a hover state
The font size test when selected should have better contrast in dark mode

### Operating System

MacOS

### AppFlowy Version(s)

Main branch

### Screenshots

_No response_

### Additional Context

_No response_

### LINKED ISSUE #2301: [Bug] Property Date in dark mode
### Bug Description

image

### How to Reproduce

Add a Date column to a table-view database
Click on the Date header

### Expected Behavior

image
 The same effect as the other two options have 

### Operating System

MacOS

### AppFlowy Version(s)

Main branch

### Screenshots

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by hyj1204**
This PR is to fix the following UI issues

### Feature Preview
**1. Update the hover effect on the `FieldTypeOptionEditor` and `FieldTypeCell`. Fix #2301** 
The updated UI 
Date cell 
image
image
Number cell
image
image

**2. Update the font size switcher UI. Fix #2299** 
Since the text color cannot be changed when it is hovered in ToggleButtons. I modified the UI to make it suitable in light&dark mode. The hover color is 30% opacity of the background color when it is selected.
In light mode
image
In dark mode
image

 **3. Improve the hover effect in `_SelectOptionColorCell`**
The hover effect for tag deletion has been updated to a high-contrast blue color in order to draw the user's attention to the action.
The hover effect for selecting a color has been updated to a light grey color. This change was made to avoid the blue color in the list from blending in with the background, as it was the same color as the previous hover effect.

image
image

**4. Improve the User setting UI**
- Resize the font size to match all the setting title
- Make the underline in the text field visible in dark mode
- Update the hover style in picking the user avatar.  The previous `IconButton` had a circle hover effect that interfered with the hover area. To fix this, I replaced it with InkWell to create a hover shape in the form of a rounded rectangle.

image

---

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/field_type_option_editor.dart [hunk 1]
```diff
@@ -113,20 +112,12 @@ class _SwitchFieldButton extends StatelessWidget {
...
       margin: GridSize.typeOptionContentInsets,
-      leftIcon: svgWidget(
-        bloc.state.field.fieldType.iconName(),
-        color: Theme.of(context).iconTheme.color,
-      ),
-      rightIcon: svgWidget(
-        "grid/more",
-        color: Theme.of(context).iconTheme.color,
-      ),
+      leftIcon: FlowySvg(name: bloc.state.field.fieldType.iconName()),
```
Comment (a-wallen) [inline frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/field_type_option_editor.dart:119]: Is there a BlocBuilder somewhere higher in the widget tree? If the state changes, but no widgets in the hierarchy rebuild in response to state changes, this state here will be stale. 

Consider using a `BlocBuilder`/`BlocListener` here if this widget needs to be used somewhere where there is no `BlocBuilder` in the widget hierarchy.

Comment (hyj1204) [reply frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/field_type_option_editor.dart:119]: You can find the `BlocBuilder` on line 43 of this file.
image
It built a `ListView` to display the editor options corresponding to the current property type(highlighted in green background). 
image

File: frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/type_option/number.dart [hunk 2]
```diff
@@ -188,7 +181,9 @@ class NumberFormatCell extends StatelessWidget {
   Widget build(BuildContext context) {
     Widget? checkmark;
```
Comment (a-wallen) [inline frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/type_option/number.dart:182]: Could you make this a getter?

```dart
Widget? get checkmark {
  if (isSelected) {
      checkmark = const FlowySvg(
        name: 'grid/checkmark',
      );
  }
  return null;
}
```

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_user_view.dart [hunk 3]
```diff
@@ -160,50 +182,45 @@ class _CurrentIcon extends StatelessWidget {
...
+            showDialog(
+              context: context,
+              builder: (BuildContext context) {
+                return SimpleDialog(
+                  title: FlowyText.medium(
+                    LocaleKeys.settings_user_selectAnIcon.tr(),
+                    fontSize: FontSizes.s16,
+                  ),
+                  children: <Widget>[
+                    SizedBox(
```
Comment (a-wallen) [inline frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_user_view.dart:208]: Is this going to be ok if we resize the window to be really small?

**Review by a-wallen — APPROVED** [review]

File: frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_user_view.dart (hunk 3 above)
Comment (hyj1204) [reply frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_user_view.dart:209]: I think we have set some minimum sizes for the window. I just tried and it looks fine.
image

File: frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/type_option/number.dart (hunk 2 above)
Comment (hyj1204) [reply frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/type_option/number.dart:182]: I think the current way is fine too. The` rightIcon` in the `FlowyButton` can be null anyway.

**Pull request merged**